logictest: add 3node-tenant directive to logictests with non-default configs

asubiotto opened the pull request:
This commit is the result of running all logictests that specify non-default
configs after applying #52680, and adding the 3node-tenant directive to tests
that pass.

For all tests with non-default configs that fail with the 3node-tenant
configuration, the failures were analyzed and marked as either expected (no
action necessary) or unexpected, in which case a 3node-tenant blocklist
directive with an associated issue was added to the list of directives.

Release note: None (testing change)

Closes #52410

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52765)
<!-- Reviewable:end -->

asubiotto commented:
CI failure was due to not being able to skip `3node-tenant`, which is a CCL-only config when run with a non-CCL binary through a normal logictest invocation. Added a commit that fixes this.

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> (although I can't really distinguish between the "expected" and "unexpected" failures of 3node-tenant config).

> Reviewed 2 of 2 files at r1, 9 of 9 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52765#-:-MEcMHSbDPQT5XIaagcL:b-hj5xvd)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 9 of 9 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52765#-:-MEcRjEeEdFMYbXvZbyw:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/logictest/logic.go, line 634 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52765#-MEcRPbABob8DNBO-AVs:-MEcRPbABob8DNBO-AVt:busu2nx) ([raw file](https://github.com/cockroachdb/cockroach/blob/be293076acd63ded932060aec144683112fb910f/pkg/sql/logictest/logic.go#L634)):*
> ```Go
> // isCCLConfig returns true for any config name that can only be run with a CCL
> // binary.
> func isCCLConfig(configName string) bool {
> ```

Consider making this a property of the `testClusterConfig` struct instead of introducing this function.


<!-- Sent from Reviewable.io -->

asubiotto reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52765)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale) (waiting on @nvanbenschoten and @yuzefovich)

___
*[pkg/sql/logictest/logic.go, line 634 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52765#-MEcRPbABob8DNBO-AVs:-MEcZALp4Yliqd798mi1:bfbq99i) ([raw file](https://github.com/cockroachdb/cockroach/blob/be293076acd63ded932060aec144683112fb910f/pkg/sql/logictest/logic.go#L634)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Consider making this a property of the `testClusterConfig` struct instead of introducing this function.
</blockquote></details>

Good point, done.


<!-- Sent from Reviewable.io -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 10 files at r3, 9 of 9 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52765#-:-MEc_u_H0NALc5cMKEQn:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

Commit:
logictest: add option to skip CCL-only configs

3node-tenant is a logictest config that can only be run with a CCL binary.
Explicitly specifying this config in a logictest directive would cause the
test to fail if run through `make testlogic`.

Previously, this was not a problem as 3node-tenant was never specified
explicitly since it was specified as an override config from the ccl package
(overrides configs if the default config would be run). However, a later
commit will add an explicit 3node-tenant directive to some tests, so this
commit adds the ability to mark a config to be run only in "CCL mode" and
adds a parameter to the logictest entry point that specifies whether the
caller wants to skip these configs or not.

Release note: None

Commit:
logictest: add 3node-tenant directive to logictests with non-default configs

This commit is the result of running all logictests that specify non-default
configs after applying #52680, and adding the 3node-tenant directive to tests
that pass.

For all tests with non-default configs that fail with the 3node-tenant
configuration, the failures were analyzed and marked as either expected (no
action necessary) or unexpected, in which case a 3node-tenant blocklist
directive with an associated issue was added to the list of directives.

Release note: None (testing change)

asubiotto commented:
bors r=yuzefovich,nvanbenschoten
